src/t_stream.c

```diff
@@ -1167,10 +1184,10 @@ int streamIteratorGetID(streamIterator *si, streamID *id, int64_t *numfields) {
             serverAssert(*numfields>=0);
 
             /* If current >= start, and the entry is not marked as
-             * deleted, emit it. */
+             * deleted or tombstones are included, emit it. */
             if (!si->rev) {
                 if (memcmp(buf,si->start_key,sizeof(streamID)) >= 0 &&
-                    !(flags & STREAM_ITEM_FLAG_DELETED))
+                    (!si->skip_tombstones || !(flags & STREAM_ITEM_FLAG_DELETED)))
```

Comment:
Whether the following `else` also need to determine `!si->skip_tombstones`.

Reply:
@itamarhaber ^

Reply:
@itamarhaber 🤦
@sundb 🙇‍♂️🙇‍♂️🙇‍♂️
Resolved via 5165321a4